ansible/roles/repo/tasks/redhat.yaml

```diff
@@ -1,16 +1,51 @@
 ---
+- set_fact:
+   rhsm_user: "{{ lookup('env', 'RHSM_USER' ) | ternary (lookup('env', 'RHSM_USER' ), rhsm_user) }}"
 
-- name: RHEL subscription
+- set_fact:
+   rhsm_password: "{{ lookup('env', 'RHSM_PASS' ) | ternary (lookup('env', 'RHSM_PASS' ), rhsm_password) }}"
+
+- set_fact:
+   rhsm_activation_key: "{{ lookup('env', 'RHSM_ACTIVATION_KEY' ) | ternary (lookup('env', 'RHSM_ACTIVATION_KEY' ), rhsm_activation_key) }}"
+
+- set_fact:
+   rhsm_activation_key: "{{ lookup('env', 'RHSM_ACTIVATION_KEY' ) | ternary (lookup('env', 'RHSM_ACTIVATION_KEY' ), rhsm_activation_key) }}"
+
+- set_fact:
+   rhsm_org_id: "{{ lookup('env', 'RHSM_ORG_ID' ) | ternary (lookup('env', 'RHSM_ORG_ID' ), rhsm_org_id) }}"
+
+- name: RHEL subscription using username and password
   redhat_subscription:
     state: present
-    username: "{{ lookup('env', 'RHSM_USER') }}"
-    password: "{{ lookup('env', 'RHSM_PASS') }}"
+    username: "{{ rhsm_user }}"
+    password: "{{ rhsm_password }}"
     auto_attach: true
     force_register: true
+    release: "{{ ansible_distribution_version }}"
+  register: rheluser
+  when:
+    - ansible_distribution == 'RedHat'
+    - rhsm_user | length > 0
+    - rhsm_password | length > 0
+
+- name: RHEL subscription using org_id and activationkey
+  redhat_subscription:
```

Comment:
Lets also update https://github.com/mesosphere/konvoy-image-builder/blob/592cefaa130c8b27ce85bf9e4d6f51205e10a690/ansible/roles/sysprep/tasks/redhat.yml#L86-L102 to use the new vars

Reply:
You should be able to use the newly registered vars `rhelorg != None or rheluser != None` to simplify the `when` condition.

Reply:
lets move this to `always`